sklearn/calibration.py

```diff
@@ -562,9 +567,26 @@ def calibration_curve(y_true, y_prob, normalize=False, n_bins=5):
     bins = np.linspace(0., 1. + 1e-8, n_bins + 1)
     binids = np.digitize(y_prob, bins) - 1
 
-    bin_sums = np.bincount(binids, weights=y_prob, minlength=len(bins))
-    bin_true = np.bincount(binids, weights=y_true, minlength=len(bins))
-    bin_total = np.bincount(binids, minlength=len(bins))
+    if sample_weight is None:
+        bin_sums = np.bincount(binids, weights=y_prob, minlength=len(bins))
+        bin_true = np.bincount(binids, weights=y_true, minlength=len(bins))
+        bin_total = np.bincount(binids, minlength=len(bins))
+    else:
+        sample_weight = check_array(sample_weight, ensure_2d=False)
+        check_consistent_length(y_true, sample_weight)
+
+        # Check that the sample weights sum is positive
+        if sample_weight.sum() <= 0:
+            raise ValueError(
+                "Attempting to calibrate predicted probabilities with a "
+                "non-positive weighted number of samples.")
+
+        bin_sums = np.bincount(binids, weights=sample_weight*y_prob,
+                               minlength=len(bins))
+        bin_true = np.bincount(binids, weights=sample_weight*y_true,
+                               minlength=len(bins))
+        bin_total = np.bincount(binids, weights=sample_weight,
+                                minlength=len(bins))
```

Comment:
please avoid the code duplication of the bincount calls. Make a weights param defined in a if else